File: src/main/java/net/moddingplayground/direbats/item/DirebatsItems.java
package net.moddingplayground.direbats.item;

import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.Item;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;
import net.moddingplayground.direbats.Direbats;

public class DirebatsItems {
    public static final Item DIREBAT_FANG = register("direbat_fang", Item::new);
    public static final Item DIREBAT_FANG_ARROW = register("direbat_fang_arrow", DirebatFangArrowItem::new);

    private static Item register(String id, Item item) {
        return Registry.register(Registry.ITEM, new Identifier(Direbats.MOD_ID, id), item);
    }

    private static Item register(String id, ItemFactory<Item> factory) {
        return register(id, factory.create(new FabricItemSettings().group(Direbats.ITEM_GROUP)));
    }

    @FunctionalInterface
    private interface ItemFactory<T extends Item> {
        T create(FabricItemSettings settings);
    }
}

File: src/main/kotlin/dev/andante/direbats/item/DirebatsItems.kt
package dev.andante.direbats.item

import dev.andante.direbats.Direbats
import dev.andante.direbats.entity.DirebatsEntityTypes
import net.minecraft.item.Item
import net.minecraft.item.SpawnEggItem
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

/**
 * Represents Direbats items.
 */
object DirebatsItems {
    /**
     * Represents a Direbat Spawn Egg item.
     */
    val DIREBAT_SPAWN_EGG = register("direbat_spawn_egg", SpawnEggItem(DirebatsEntityTypes.DIREBAT, 0x3E3E3E, 0xF8E1D3, Item.Settings().group(DirebatsItemGroups.ALL)))

    /**
     * Represents a Direbat Fang item.
     */
    val DIREBAT_FANG = register("direbat_fang", ::Item)

    /**
     * Represents a Direbat Fang Arrow item.
     */
    val DIREBAT_FANG_ARROW = register("direbat_fang_arrow", ::DirebatFangArrowItem)

    private fun register(id: String, item: Item): Item {
        return Registry.register(Registry.ITEM, Identifier(Direbats.MOD_ID, id), item)
    }

    private fun register(id: String, item: (Item.Settings) -> Item): Item {
        return register(id, item.invoke(Item.Settings().group(DirebatsItemGroups.ALL)))
    }
}
